cmd/cluster-client-gen/generators/client_generator.go

```diff
@@ -203,7 +203,7 @@ func targetForGroup(gv clientgentypes.GroupVersion, typeList []*types.Type, clie
 
 func targetForClientset(args *args.Args, clientsetDir, clientsetPkg string, groupGoNames map[clientgentypes.GroupVersion]string, boilerplate []byte) generator.Target {
 	return &generator.SimpleTarget{
-		PkgName:       "clientset",
+		PkgName:       path.Base(clientsetPkg),
```

Comment:
Could also make this an input, e.g. `clientsetPkgName` but imho thats just another possible error source and `path.Base` should always return the correct value.

Reply:
Just make sure this works both in the code-generator's examples and for kcp itself. Both have widely different directory structures, so assumptions are difficult to make here 😁

Reply:
I had only noticed this because the tests in kcp were complaining about the mismatch :D